src/engine.ml

```diff
@@ -95,13 +92,13 @@ let compute_hmac key p hmac_algorithm hmac_key =
   let tbs = Packet.to_be_signed key p in
   Mirage_crypto.Hash.mac hmac_algorithm ~key:hmac_key tbs
 
-let hmac_and_out protocol key { hmac_algorithm; my_hmac; _ } (p : Packet.pkt) =
+let hmac_and_out protocol { hmac_algorithm; my_hmac; _ } ((key, p) : Packet.t) =
   let hmac = compute_hmac key p hmac_algorithm my_hmac in
   let header = Packet.header p in
   let p' = Packet.with_header { header with Packet.hmac } p in
   Packet.encode protocol (key, p')
 
-let encrypt_and_out protocol key { my_key; my_hmac; _ } (p : Packet.pkt) =
+let encrypt_and_out protocol { my_key; my_hmac; _ } ((key, p) : Packet.t) =
```

Comment:
We need to ensure we don't call this with ``(_, `Data _)`` as `Packet.Tls_crypt.encode` will assert false.